### ISSUE TITLE
M2.1.3 : Form validation with identical field names does not work as expected.

### ISSUE THREAD

**Issue by kanduvisla** [magento/magento2#8258]
When you have multiple fields that share the same name, the JavaScript form validation does not work.

### Preconditions

1. Magento 2.1.3

### Steps to reproduce

Create a form with identical fields and add JavaScript validation. For example:

```
<form id="form-validate">
    <input type="text" name="street[]" class="required-entry" />
    <input type="text" name="street[]" class="required-entry" />
    <input type="text" name="street[]" class="required-entry" />
    <input type="submit" />
</form>
<script type="text/x-magento-init">
    {
        "#form-validate": {
            "validation": {}
        }
    }
</script>
```

### Expected result

When trying to submit the form, all 3 fields should show an error message according to their validation rule(s).

### Actual result

The message is only shown for the first field. Furthermore, the other fields are not validated at all.

**Comment by Danielc3** [comment]
Any news on this? This is also an issue for me.

**Comment by gautamkrishnar** [comment]
Any update on this one?

**Comment by magento-engcom-team** [comment]
@kanduvisla, thank you for your report.
We've created internal ticket(s) MAGETWO-81885 to track progress on the issue.

**Comment by jayankaghosh** [comment]
I am working on this at #dmcdindia

**Comment by Arun-smacon** [comment]
I am working on this at #dmcdindia

**Comment by magento-engcom-team** [comment]
@Arun-smacon thank you for joining. Please accept team invitation [here](https://github.com/orgs/magento/invitation) and self-assign the issue.

**Comment by gautamkrishnar** [comment]
@jayankaghosh @Arun-smacon 👍 🎉 All the best guys 👏

**Comment by jayankaghosh** [comment]
Has been fixed in the above mentioned commit. Tested on 2.1.3 and it's working

**Comment by magento-engcom-team** [comment]
Hi @kanduvisla. Thank you for your report. 
The issue has been fixed in magento/magento2#15383 by @jayankaghosh in 2.3-develop branch
Related commit(s):
 - [113440fde52703e35e9c80e0c2229c91b02a3222](https://github.com/magento/magento2/commit/113440fde52703e35e9c80e0c2229c91b02a3222)
 - [874d77083329e66ea574ad32dc1d98184e45e71a](https://github.com/magento/magento2/commit/874d77083329e66ea574ad32dc1d98184e45e71a)
 - [84355a34d260f5720af626221cbbac36a6533b73](https://github.com/magento/magento2/commit/84355a34d260f5720af626221cbbac36a6533b73)
 - [d5c41b643b3b8d92a66da633a0fdb8a2312cbd86](https://github.com/magento/magento2/commit/d5c41b643b3b8d92a66da633a0fdb8a2312cbd86)
 - [156ac1654294f7b0e24057cf10ce2e098b354c7f](https://github.com/magento/magento2/commit/156ac1654294f7b0e24057cf10ce2e098b354c7f)
 - [65a2c17418755355bc50dd7aded5a6bd617c97dc](https://github.com/magento/magento2/commit/65a2c17418755355bc50dd7aded5a6bd617c97dc)
 - [97c03f5deb4ee53a1a597610509f9ecf91c029ff](https://github.com/magento/magento2/commit/97c03f5deb4ee53a1a597610509f9ecf91c029ff)
 - [224f73768ea425fcd8128f921682a5aa894e5992](https://github.com/magento/magento2/commit/224f73768ea425fcd8128f921682a5aa894e5992)
 - [908100bdc9974d3754ad3291763705ee3ff321c1](https://github.com/magento/magento2/commit/908100bdc9974d3754ad3291763705ee3ff321c1)
 - [38b9709750b6d8577afaecb95b5b8cedc7425a59](https://github.com/magento/magento2/commit/38b9709750b6d8577afaecb95b5b8cedc7425a59)
 - [388d31d1993378740957236b15e3ce27733748e2](https://github.com/magento/magento2/commit/388d31d1993378740957236b15e3ce27733748e2)
 - [ee288c10b9edfa02557c046efc1f1476d5c60808](https://github.com/magento/magento2/commit/ee288c10b9edfa02557c046efc1f1476d5c60808)
 - [eadcff2fd8bfb6a92ea371232fc0bde562f39552](https://github.com/magento/magento2/commit/eadcff2fd8bfb6a92ea371232fc0bde562f39552)

The fix will be available with the upcoming 2.3.3 release.
